### PR TITLE
Prevent detection when busy

### PR DESCRIPTION
**Pull request by haslinghuis**
Prevent auto-detection during verify board and backup on flash as it might disturb other processes.

Excluded is to make the button smarter as it does have lot's of events

enable autodetect when

- initialize tab
- new port is detected
- board is switched and was not same target
- abort flashing after flash firmware is cancelled

disable autodetect when

- in DFU mode
- board is switched and was same target
- using autodetect to verify board (which triggers port change)
- using backup on flash (which also triggers port change)
- load firmware

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by nerdCopter** [comment]
problem/solution description?

**Comment by nerdCopter** [comment]
- working pretty good. seemingly better than master, but not too deeply tested.
- flashed 2 different FC a few times with a few different defines.
- autodetect working well when swapping plugged FC's.
- skips for DFU mode as expected.
- skips autodetect on same FC as expected.

**Review by chmelevskij — APPROVED** [review]

**Comment by nerdCopter** [comment]
`e09d7872`
- does not auto-detect if FC is plugged before launching configurator (maybe by design & not a problem for me).
- auto-detects upon FC plugin.
- skips auto-detect for same FC replug.
- auto-detects upon different FC plugin.
- flashing works.

it's really good. impressive improvements.

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**
